pkg/testutils/lint/passes/fmtsafe/functions.go

```diff
@@ -85,6 +91,14 @@ var requireConstFmt = map[string]bool{
 	"(*github.com/cockroachdb/cockroach/pkg/util/grpcutil.grpcLogger).Errorf":   true,
 	"(*github.com/cockroachdb/cockroach/pkg/util/grpcutil.grpcLogger).Fatalf":   true,
 
+	// Both of these signatures need to be included for the linter to not flag
+	// roachtest testImpl.Errorf since it is in the main package
+	"(*main.testImpl).Errorf": true,
```

Comment:
This is quite unfortunate, as it seems like it's the result of a bug somewhere (we shouldn't be having to define the `*main` part of this). I spent some time looking into this and I suspect it's an issue with `nogo` (potentially only in our fork).

Reply:
`nogo` just runs `checkPackage`; something must be up with the passed `packagePath` (aka `importpath`). It seems correct in `BUILD.bazel`. I wonder what might be passing `main` instead?

```
importpath = "github.com/cockroachdb/cockroach/pkg/cmd/roachtest/"
```

Reply:
Methinks this might be the culprit [1]. It's hardcoded `importpath` to `main` from the `archive` action which calls into `compilePkg`, and then into `checkPackage`.

[1] https://github.com/bazelbuild/rules_go/pull/2135